core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -5914,6 +5866,73 @@ enum DefaultConstants NADefaults::validateAndInsert(const char *attrName,
 
 } // NADefaults::validateAndInsert()
 
+float NADefaults::computeNumESPsPerCore(NABoolean aggressive)
+{
+   #define DEFAULT_ESPS_PER_NODE 2   // for conservation allocation
+   #define DEFAULT_ESPS_PER_CORE 0.5 // for aggressive allocation
+
+     // Make sure the gpClusterInfo points at an NAClusterLinux object.
+     // In osim simulation mode, the pointer can point at a NAClusterNSK
+     // object, for which the method numTSEsForPOS() is not defined.
+   NAClusterInfoLinux* gpLinux = dynamic_cast<NAClusterInfoLinux*>(gpClusterInfo);
+   assert(gpLinux);		
+
+   // cores per node
+   Lng32 coresPerNode = gpClusterInfo->numberOfCpusPerSMP();
+
+   if ( aggressive ) {
+      float totalMemory = gpLinux->totalMemoryAvailable(); // per Node, in KB
+      totalMemory /= 1024; // per Node, in GB
+      totalMemory /= coresPerNode ; // per core, in GB
+      totalMemory /= 2; // per core, 2GB per ESP
```

Comment:
Maybe wait until we receive some data on the default setting?

On Wed, Jan 6, 2016 at 5:18 PM, Hans Zeller <notifications@github.com>
wrote:

> In core/sql/sqlcomp/nadefaults.cpp
> <https://github.com/apache/incubator-trafodion/pull/250#discussion_r49023709>
> :
>
> > +   #define DEFAULT_ESPS_PER_CORE 0.5 // for aggressive allocation
> > +
> > +     // Make sure the gpClusterInfo points at an NAClusterLinux object.
> > +     // In osim simulation mode, the pointer can point at a NAClusterNSK
> > +     // object, for which the method numTSEsForPOS() is not defined.
> > +   NAClusterInfoLinux* gpLinux = dynamic_cast<NAClusterInfoLinux*>(gpClusterInfo);
> > +   assert(gpLinux);		
> > +
> > +   // cores per node
> > +   Lng32 coresPerNode = gpClusterInfo->numberOfCpusPerSMP();
> > +
> > +   if ( aggressive ) {
> > +      float totalMemory = gpLinux->totalMemoryAvailable(); // per Node, in KB
> > +      totalMemory /= 1024; // per Node, in GB
> > +      totalMemory /= coresPerNode ; // per core, in GB
> > +      totalMemory /= 2; // per core, 2GB per ESP
>
> Should this be another CQD?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/250/files#r49023709>.
>



-- 
Regards, --Qifan